src/Task/TalkNPC.pm

```diff
@@ -619,51 +640,28 @@ sub iterate {
 		}
 		
 		# No more steps to be sent
-		if ($self->noMoreSteps) {
-			# Usual end of a conversation
-			if (!%talk) {
-				$self->conversation_end;
-			
-			# No more steps but we still have a defined %talk
-			} else {
-				debug "Done talking with $self->{target}, but another NPC initiated a talk instantly\n", 'ai_npcTalk';
-				# TODO: maybe better create a new task
-				debug "Continuing the talk within the same task\n", 'ai_npcTalk';
-				my $target = $self->find_and_set_target;
+		# Usual end of a conversation
+		if ($self->noMoreSteps && !%talk) {
+			$self->conversation_end;
+		
+		# There are more steps but no conversation with npc
+		} elsif (!%talk) {
```

Comment:
it won't reinitiate the conversation unless the next step is x, it will just print an error.